ludwig/utils/triton_utils.py

```diff
@@ -49,142 +93,522 @@ def forward(self, {input_signature}):
+            f.write(self.get_config())
+        return config_path
 
-def _get_instance_spec(count=1, kind="KIND_CPU") -> str:
-    spec = INSTANCE_SPEC.format(count=count, kind=kind)
-    return spec
+    def save_ensemble_dummy_model(self) -> str:
+        """Scripts the model and saves it."""
+        if not isinstance(self.model_version, int) or self.model_version < 1:
+            raise ValueError("Model version has to be a non-zero positive integer")
+        pass
 
+        os.makedirs(os.path.join(self.base_path, str(self.model_version)), exist_ok=True)
+        model_path = os.path.join(self.base_path, str(self.model_version), "model.txt")
+        with open(model_path, "w") as f:
+            f.write("no model for the ensemble")
 
-def _get_model_config(model: LudwigModel) -> str:
-    config = TRITON_CONFIG_TEMPLATE.format(
-        input_spec=_get_input_spec(model), output_spec=_get_output_spec(model), instance_spec=_get_instance_spec()
-    )
-    return config
+        return model_path
+
+
+@dataclass
+class TritonConfig:
+    """Enables the creation and export of a Triton config.
+
+    :param full_model_name: name of the model. Must be the same as the directory where the config is saved.
+    :param input_features: input features of the model.
+    :param output_features: output features of the model.
+    """
+
+    full_model_name: str
+    input_features: List[TritonConfigFeature]
+    output_features: List[TritonConfigFeature]
+    device: str
+    device_count: int
+    inference_stage: str
+
+    def _get_triton_spec(self, triton_features: List[TritonConfigFeature]) -> str:
+        spec = []
+        for feature in triton_features:
+            spec.append(
+                TRITON_SPEC.format(
+                    key=feature.key,
+                    name=feature.name,
+                    data_type=feature.type,
+                    data_dims=", ".join(str(dim) for dim in feature.dimension),  # check correctness
+                    reshape_spec=self._get_reshape_spec(feature),
+                )
+            )
+        return ",".join(spec)
+
+    def _get_reshape_spec(self, feature):
+        if feature.kind == INPUT and self.inference_stage == PREDICTOR:
+            return FEATURE_RESHAPE_SPEC.format(reshape_dims=", ".join(str(dim) for dim in feature.dimension[1:]))
+        return ""
+
+    def _get_instance_spec(self) -> str:
+        if self.device == "cpu":
+            kind = "KIND_CPU"
+        else:
+            kind = "KIND_GPU"
+        spec = INSTANCE_SPEC.format(count=self.device_count, kind=kind)
+        return spec
+
+    def _get_dynamic_batching_spec(self):
+        if self.inference_stage == PREDICTOR:
+            return DYNAMIC_BATCHING_TEMPLATE
+        return ""
+
+    def get_model_config(self) -> str:
+        """Generate a Triton config for a model from the input and output features."""
+        max_batch_size = 0
+        if self.inference_stage == PREDICTOR:
+            max_batch_size = 1
```

Comment:
Should we add an argument to allow passing this in, given that we would probably want this to be more than 1 in most cases (perhaps we could default to 64 if we don't get a value?)